detect.go

```diff
@@ -0,0 +1,34 @@
+package dotnetcoreaspnetruntime
+
+import "github.com/paketo-buildpacks/packit/v2"
+
+type Environment struct {
+	DotnetRollForward      string `env:"BP_DOTNET_ROLL_FORWARD"`
+	DotnetFrameworkVersion string `env:"BP_DOTNET_FRAMEWORK_VERSION"`
+}
+
+func Detect(environment Environment) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		var requirements []packit.BuildPlanRequirement
+
+		if environment.DotnetFrameworkVersion != "" {
+			requirements = append(requirements, packit.BuildPlanRequirement{
+				Name: "dotnet-core-aspnet-runtime",
+				Metadata: map[string]interface{}{
+					"version-source": "BP_DOTNET_FRAMEWORK_VERSION",
+					"version":        environment.DotnetFrameworkVersion,
+					"launch":         true,
```

Comment:
Why specifically required at launch in this instance?

Reply:
I looked at a couple of other buildpacks and we set launch true for the launch bit if the version variable is set so I just went with that idea.

Reply:
I know we'll want it at launch but isn't that the business of other buildpacks?

Reply:
Hmmm just took a look at the buildpacks I thought I saw this behavior in and I am not seeing it. Must be a misremembering I will fix it.